PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx

```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
 const double ptDAxisMin = 0.;
 const double ptDAxisMax = 36.;
 
-using MCParticlesPlus3Prong = soa::Join<aod::McParticles, aod::HfCand3ProngMcGen>;
+// definition of ME variables
+std::vector<double> zBins{VARIABLE_WIDTH, -10.0, -2.5, 2.5, 10.0};
+std::vector<double> multBins{VARIABLE_WIDTH, 0., 200., 500.0, 5000.};
+std::vector<double> multBinsMcGen{VARIABLE_WIDTH, 0., 20., 50.0, 500.}; // In MCGen multiplicity is defined by counting primaries
+using BinningType = ColumnBinningPolicy<aod::collision::PosZ, aod::mult::MultFV0M<aod::mult::MultFV0A, aod::mult::MultFV0C>>;
+BinningType corrBinning{{zBins, multBins}, true};
+
+// Code to select a Dmeson in a collision
+struct HfDplusSelection {
+  Produces<aod::DmesonSelection> dplusSel;
+  Configurable<int> selectionFlagDplus{"selectionFlagDplus", 1, "Selection Flag for Dplus"};
+  Configurable<double> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<double> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>> selectedDPlusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= selectionFlagDplus;
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec>> recoFlagDPlusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi > 0;
+
+  void processDplusSelectionData(aod::Collision const& collision, soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi> const& candidates)
+  {
+    int isDplusFound = 0;
+    if (selectedDPlusCandidates.size() > 0) {
+      auto selectedDPlusCandidatesGrouped = selectedDPlusCandidates->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex());
+
+      for (auto& candidate1 : selectedDPlusCandidatesGrouped) {
+        if (yCandMax >= 0. && std::abs(yDplus(candidate1)) > yCandMax) {
+          continue;
+        }
+        if (ptCandMin >= 0. && candidate1.pt() < ptCandMin) {
+          continue;
+        }
+        isDplusFound = 1;
+      }
```

Comment:
You can already exit the loop at this point.

Reply:
break is already done in the loop.